tests/test_uncrustify/test.py

```diff
@@ -108,18 +112,30 @@ def run(self, args):
         except subprocess.CalledProcessError as exc:
             msg = '{}: Uncrustify error code {}'
             msg = msg.format(self.test_name, exc.returncode)
-            printc('FAILED: ', msg, **FAIL_ATTRS)
-            raise ExecutionFailure(exc)
+            if not self.test_xfail:
+                printc('FAILED: ', msg, **FAIL_ATTRS)
+                raise ExecutionFailure(exc)
+            else:
+                printc('XFAILED: ', msg, **FAIL_ATTRS)
+                return
         finally:
             del stderr
 
         try:
-            if not filecmp.cmp(_expected, _result):
```

Comment:
Thinking about the previous comment, I think XFAIL diffs should not be printed "by default". We could add a separate option, but I think it would make sense to print XFAIL diffs iff both `-d` (`args.diff`) and `-p` (`args.show_all`) are passed?

Reply:
for `XFAIL` you just know that there will be a diff, probably different every time uncrustify changes, but still not the output you are expecting so they are kinda irrelevant. If we find the diff relevant we can always print it on some option.